### PR TITLE
Remove hardcoded panel sizes in launch game window

### LINKED ISSUE #5631: Launch Game window does not resize contents properly
**Describe the bug**
The "Launch Game" window can increase the size of some parts, but not decrease them when the game window is resized.

**To Reproduce**
1. Start new single player game and select map to get to the "Launch Game" window
2. Resize the game window by dragging by the edges so that only the width or only the height changes. Try increasing then decreasing the size
3. See error

**Expected behavior**
All window elements should be resized properly for all changes

**Screenshots**
shot0026
shot0027
shot0028
shot0029
shot0030
shot0031

**Version:**
 - OS: antix/Debian buster with backports
 - Widelands Version: v1.1

**Additional context**
Multiplayer launch window only has problem with vertical resize. If both width and height change (dragging by window corner), then the contents are resized properly. Also if e.g. width is decreased, when height is changed afterwards, the widths will be resized properly.

### PR DESCRIPTION
**Pull request by matthiakl**
**Type of change**
Bugfix

**Issue(s) closed**
Fixes #5631 

**How it works**
If the desired size of a panel overflows the box there is currently no check to prevent this. The short term fix for the bug is to reduce the desired sizes of the problematic panels. The long term fix would be to sanitize box layouts to not overflow in general.

**Additional context**
I already tested a bit of a generic overflow detection, but this is no trivial task because of the `set_size() -> layout() -> set_size()` cycles.

### CODE REVIEW

**Comment by Noordfrees** [comment]
🎉 

The bug still happens though with the MP setup window at 1024×768 when clicking the "Show More Options" button:
grafik


Also, the label for the win condition duration spinner is still cut off if the window is narrow. That's not necessarily in scope of this PR but #5631 isn't truly fixed before this is also solved somehow.

**Comment by matthiakl** [comment]
Should be all fixed now:
- The spinbox label is hidden when the space is not sufficient
- The advanced options are in a scrollable box, it is a bit cramped, but does not overflow anymore.

File: src/ui_basic/box.cc [hunk 1]
```diff
@@ -291,6 +285,18 @@ void Box::update_positions() {
...
 		}
 	}
+
+#ifndef NDEBUG
+	// Subtract last spacing + some leeway from infspace calculation
+	totaldepth = totaldepth - items_.size() - inner_spacing_;
+	int max_depths = orientation_ == Horizontal ? get_w() : get_h();
+	if (!force_scrolling_ && !scrolling_ && max_depths > 0 && totaldepth > max_depths) {
+		// TODO(matthiakl): one day this should be an error
+		verb_log_warn("Overflowing %s box %s: %d > %d ",
```
Comment (Noordfrees) [inline src/ui_basic/box.cc]:
```suggestion
		verb_log_dbg("Overflowing %s box %s: %d > %d ",
```

**Review by Noordfrees — COMMENTED** [review]
LGTM, works well :)
One request, the label for the spinbox shouldn't disappear completely, could you make it a tooltip instead when not shown (or maybe always)?

**Review by Noordfrees — APPROVED** [review]
Great, thanks :)

**Pull request merged**